nvflare/private/fed/server/job_cmds.py

```diff
@@ -407,12 +407,20 @@ def abort_job(self, conn: Connection, args: List[str]):
         try:
             job_id = conn.get_prop(self.JOB_ID)
             with engine.new_context() as fl_ctx:
-                message = job_runner.stop_run(job_id, fl_ctx)
-                if message:
-                    conn.append_error(message, meta=make_meta(MetaStatusValue.INTERNAL_ERROR, message))
-                else:
-                    conn.append_string("Abort signal has been sent to the server app.")
+                job_manager = engine.get_component(SystemComponents.JOB_MANAGER)
+                job = job_manager.get_job(job_id, fl_ctx)
+                if job.meta.get(JobMetaKey.STATUS) in [RunStatus.SUBMITTED, RunStatus.DISPATCHED]:
+                    job_manager.set_status(job.job_id, RunStatus.FINISHED_ABORTED, fl_ctx)
+                    conn.append_string(f"Abort the job: {job_id} before run.")
                     conn.append_success("", make_meta(MetaStatusValue.OK))
+                    return
+                else:
```

Comment:
This already handled. For those FINISHED:xxx job, it will result in the following error message:

Error: Job xxxxx is not running.

Reply:
No, we shouldn't rely on job_runner.stop_run() to tell us that, especially when the status is already FINISHED:XXX, we shouldn't even have to call it.

The returned message is a str for the user to read, but hard for flare_api to digest - it only reads stuff in meta.

We should explicitly check FINISHED: status, and then:
conn.append_string(f"Job {job_id} is not running", meta=make_meta(MetaStatusValue.OK, "job not running"))
The important thing here is MetaStatusValue.OK; otherwise flare_api would raise INTERNAL_ERROR exception.

Reply:
done.